src/peripherals/stm_flash.c

```diff
@@ -117,6 +121,26 @@ u8 stm_flash_program(u32 address, u8 data[], u8 length)
   return FLASH_OK;
 }
 
+/** Callback to read STM32F4's hardcoded unique ID.
+ * Sends STM32F4 unique ID (12 bytes) back to host.
+ */
+void stm_unique_id_callback(u16 sender_id, u8 len, u8 msg[], void* context)
```

Comment:
As a general principle I think its good to keep the things under `board/` and `peripheral/` factored out from the rest of the FW as much as possible.

They are kind of like mini libraries for interacting with certain subsystems. I think its much less confusing and more maintainable if they do not use SBP etc. directly. Its the job of the firmware to combine those libraries with the SBP or whatever functionality.